magento/pwa-studio
# chore: Allow for async taps of targets

## Description

It may come to be that we need to perform some async calls to determine available routes. This change makes that possible, in a backwards compatible way that doesn't break anyone already tapping into the available targets.

```js
targets.own.routes.tapPromise(async routes => {
    const builtinRoutes = [
        {
            name: 'SavedPaymentsPage',
            pattern: '/saved-payments',
            exact: true,
            path: '../SavedPaymentsPage'
        },
        // ... other built in routes
    ];

    // Returns ['SavedPaymentsPage', ...] based on GQL introspection.
    const disabledRoutes = await getDisabledRoutes();

    return [...routes, ...builtinRoutes].filter(route =>
        disabledRoutes.contains(route.name)
    );
});
```

## Related Issue





Closes [PWA-933](https://github.com/magento/pwa-studio/pull/2718).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Run `yarn build` and make sure it completes without error.
2. Change line in venia-ui-intercept.js to `targets.own.routes.tapPromise(async routes => {` and run `yarn build` and it should complete without error.
3. `yarn watch:venia` and make sure you can hit the UI without error.
4. `yarn stage:venia` and make sure you can hit the UI without error.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
